scripts/azure/storage-manager.sh

```diff
@@ -0,0 +1,157 @@
+
+# shellcheck disable=SC2016
+wc_default=$(echo "${common_default}" | yq4 eval-all --prettyPrint '. as $item ireduce ({}; . * $item )' - <(yq4 -o j '.objectStorage // {}' "${CK8S_CONFIG_PATH}/defaults/wc-config.yaml"))
+# shellcheck disable=SC2016
+wc_config=$(echo "${wc_default}" | yq4 eval-all --prettyPrint '. as $item ireduce ({}; . * $item )' - <(yq4 -o j '.objectStorage // {}' "${CK8S_CONFIG_PATH}/common-config.yaml") <(yq4 -o j '.objectStorage // {}' "${CK8S_CONFIG_PATH}/wc-config.yaml") |  yq4 '{"objectStorage":.}' -)
+
+objectstorage_type_sc=$(echo "${sc_config}" | yq4 '.objectStorage.type' -)
+objectstorage_type_wc=$(echo "${wc_config}" | yq4 '.objectStorage.type' -)
+
+[ "$objectstorage_type_sc" != "azure" ] && log_info "Azure Storage is not enabled in service cluster"
+[ "$objectstorage_type_wc" != "azure" ] && log_info "Azure Storage is not enabled in workload cluster"
+
+if [ "$objectstorage_type_sc" != "azure" ] && [ "$objectstorage_type_wc" != "azure" ]; then
+    log_error "Azure Storage is not enabled in either cluster, aborting!"
+    exit 1
+fi
+
+[ "$objectstorage_type_sc" = "azure" ] && buckets_sc=$(echo "${sc_config}" | yq4 '.objectStorage.buckets.*' -)
+[ "$objectstorage_type_wc" = "azure" ] && buckets_wc=$(echo "${wc_config}" | yq4 '.objectStorage.buckets.*' -)
+
+CONTAINERS=$( { echo "$buckets_sc"; echo "$buckets_wc"; } | sort | uniq | tr '\n' ' ' | sed s'/.$//')
+
+log_info "Operating on containers: ${CONTAINERS// /', '}"
+
+function usage() {
+    echo "Usage:" 1>&2
+    echo " $0 create" 1>&2
+    echo " $0 delete" 1>&2
+}
+
+case "$1" in
+create)
+    ACTION=$CREATE_ACTION
+    ;;
+delete)
+    ACTION=$DELETE_ACTION
+    ;;
+esac
+shift
+
+function create_resource_group() {
+
+    log_info "checking if resource group exists" >&2
+    GROUP_EXISTS=$(az group list --query '[].name' | jq --arg group "${CK8S_ENVIRONMENT_NAME}"-storage-resource-group '. | index($group)')
+    if [ "$GROUP_EXISTS" != null ]; then
+        log_info "resource group [${CK8S_ENVIRONMENT_NAME}-storage-resource-group] already exists" >&2
+        log_info "continue using this group ? (y/n)" >&2
+        read -r -n 1 cmdinput
+        case "$cmdinput" in
+        y|Y) return ;;
+        *) exit 0 ;;
+        esac
+    else
+        log_info "resource group [${CK8S_ENVIRONMENT_NAME}-storage-resource-group] does not exist, creating it now" >&2
+        az group create \
+            --name "$CK8S_ENVIRONMENT_NAME"-storage-resource-group \
+            --location "${AZURE_LOCATION}" --only-show-errors
+    fi
+}
+
+function create_storage_account() {
+
+    log_info "checking storage account availability" >&2
+    out=$(az storage account check-name --only-show-errors --name "${CK8S_ENVIRONMENT_NAME}"storageaccount)
```

Comment:
The storage account should be read from the config or secrets, and set during init.